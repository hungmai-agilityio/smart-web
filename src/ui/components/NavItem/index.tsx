'use client';
import Link from 'next/link';

interface NavItemProps {
  name: string;
  url?: string;
}

export const NavItem = ({ name, url }: NavItemProps) => {
  return (
    <Link href={url || '#'}>
      <div className="flex items-center mb-3 relative cursor-pointer">
        <p className="font-bold lg:text-lg text-sm hover:text-second uppercase">
          {name}
        </p>
      </div>
    </Link>
  );
};
