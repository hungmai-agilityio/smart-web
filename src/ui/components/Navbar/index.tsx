import { NavItem } from '@/ui/components/Navbar/Item';

interface NavbarProps {
  items: Array<{ name: string; url?: string }>;
  styles?: string;
}

export const Navbar = ({ items, styles }: NavbarProps) => {
  return (
    <nav className={styles}>
      {items.map((item, index) => (
        <NavItem key={index} name={item.name} url={item.url} />
      ))}
    </nav>
  );
};
