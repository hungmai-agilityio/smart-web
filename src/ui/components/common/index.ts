export * from '@/ui/components/common/Button';
export * from '@/ui/components/common/Dropdown';
export * from '@/ui/components/common/Icon';
export * from '@/ui/components/common/Photo';
export * from '@/ui/components/common/Text';
export * from '@/ui/components/common/StatsRow';
