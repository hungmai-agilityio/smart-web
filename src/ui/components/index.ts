export * from '@/ui/components/Hero';
export * from '@/ui/components/NavItem';
export * from '@/ui/components/Portfolio';
export * from '@/ui/components/Pagination';
export * from '@/ui/components/Skeleton/Card';
export * from '@/ui/components/Skeleton/Image';
export * from '@/ui/components/common';
