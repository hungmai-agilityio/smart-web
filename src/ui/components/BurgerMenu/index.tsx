'use client';

import Image from 'next/image';
import { useRef, useState } from 'react';
import { Navbar } from '@/ui/components';
import { useOutsideClick } from '@/hooks';

interface BurgerProps {
  items: Array<{ name: string; url?: string }>;
}

export const BurgerMenu = ({ items }: BurgerProps) => {
  const menuRef = useRef<HTMLDivElement>(null);
  const [menuOpen, setMenuOpen] = useState<boolean>(false);

  const toggleMenu = () => {
    setMenuOpen(!menuOpen);
  };

  useOutsideClick({
    ref: menuRef,
    handler: () => setMenuOpen(false)
  });

  return (
    <div ref={menuRef}>
      <div
        className="md:hidden flex items-center cursor-pointer"
        onClick={toggleMenu}
      >
        <Image src="/menu.svg" alt="menu" width={16} height={14} priority />
      </div>

      <Navbar
        items={items}
        styles={`md:flex gap-7 ${menuOpen ? 'block' : 'hidden'}
          md:relative absolute top-full left-0 w-full md:w-auto z-10 transition-all ease-in-out
          bg-white p-5 drop-shadow-lg md:drop-shadow-none lg:mr-28 md:mr-8`}
      />
    </div>
  );
};
