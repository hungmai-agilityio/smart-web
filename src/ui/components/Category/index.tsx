'use client';

import { memo, useCallback } from 'react';
import { useRouter, useSearchParams } from 'next/navigation';

// Constants
import { abel } from '@/constants';

// Components
import { Dropdown, ItemProps } from '@/ui/components';

interface CategoryProps {
  categories: string[];
  queryParam: string;
}

export const Category = memo(({ categories, queryParam }: CategoryProps) => {
  const router = useRouter();
  const searchParams = useSearchParams();

  const currentCategory = searchParams.get(queryParam);

  // Update category and URL when a new category is selected
  const handleCategorySelect = useCallback(
    (category: string) => {
      const params = new URLSearchParams(searchParams.toString());
      params.set(queryParam, category);
      router.push(`?${params.toString()}`, { scroll: false });
    },
    [searchParams, router, queryParam]
  );

  const dropdownItems: ItemProps[] = categories.map((category) => ({
    label: category,
    value: category
  }));

  return (
    <>
      <ul
        className={`${abel.className} my-8 md:flex hidden lg:gap-48 gap-16 justify-center`}
      >
        {categories.map((category, index) => (
          <li
            className={`uppercase cursor-pointer hover:text-second ${currentCategory === category ? 'text-second' : ''}`}
            key={index}
            onClick={handleCategorySelect.bind(null, category)}
          >
            {category}
          </li>
        ))}
      </ul>
      <div className="md:hidden flex justify-center my-8">
        <Dropdown
          items={dropdownItems}
          selectedValue={currentCategory!}
          onSelect={(value) => handleCategorySelect(value)}
        />
      </div>
    </>
  );
});
