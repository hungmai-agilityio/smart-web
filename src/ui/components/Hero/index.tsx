import { ReactNode } from 'react';

// Constants
import { TYPE } from '@/constants';

// Components
import { Button, Photo } from '@/ui/components/common';

interface HeroProps {
  imageUrl: string;
  children?: ReactNode;
  buttonName?: string;
}

export const Hero = ({ imageUrl, children, buttonName }: HeroProps) => {
  return (
    <div className="relative w-full h-hero">
      <Photo src={imageUrl} alt="Hero Background" autoSize priority />
      <div className="absolute  inset-0 flex flex-col items-center justify-center">
        {children}
        {buttonName && (
          <div className="mt-10">
            <Button variant={TYPE.PRIMARY} name={buttonName} />
          </div>
        )}
      </div>
    </div>
  );
};
