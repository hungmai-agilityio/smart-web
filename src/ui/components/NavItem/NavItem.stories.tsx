import { NavItem } from '@/ui/components/NavItem';
import type { Meta, StoryObj } from '@storybook/react';

const meta: Meta<typeof NavItem> = {
  component: NavItem,
  tags: ['autodocs'],

  parameters: {
    layout: 'centered',
    nextjs: {
      appDirectory: true
    }
  },

  argTypes: {
    name: {
      description: 'Name for the navigation bar item'
    },
    url: {
      description: 'path to go to a certain page '
    }
  }
};

export default meta;

type Story = StoryObj<typeof NavItem>;

export const Default: Story = {
  args: {
    name: 'About',
    url: '/about'
  }
};
