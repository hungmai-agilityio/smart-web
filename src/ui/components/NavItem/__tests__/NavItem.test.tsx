import { render, screen } from '@testing-library/react';
import { NavItem } from '@/ui/components/NavItem';
import { fireEvent } from '@testing-library/react';

describe('NavItem', () => {
  const navName = 'home';
  const navUrl = '/home';

  it('Should render value passed in props', () => {
    render(<NavItem name={navName} />);

    expect(screen.getByText(navName)).toBeInTheDocument();
  });

  it('Should trigger navigation when clicked', () => {
    render(<NavItem name={navName} url={navUrl} />);

    const linkElement = screen.getByText(navName);
    fireEvent.click(linkElement);

    expect(linkElement.closest('a')).toHaveAttribute('href', navUrl);
  });

  it('Should match snapshot', () => {
    const { container } = render(<NavItem name={navName} url={navUrl} />);
    expect(container).toMatchSnapshot();
  });
});
