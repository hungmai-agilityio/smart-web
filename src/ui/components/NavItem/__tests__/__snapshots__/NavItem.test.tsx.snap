// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavItem Should match snapshot 1`] = `
<div>
  <a
    href="/home"
  >
    <div
      class="flex items-center mb-3 relative cursor-pointer"
    >
      <p
        class="font-bold lg:text-lg text-sm hover:text-second uppercase"
      >
        home
      </p>
    </div>
  </a>
</div>
`;
